Lib/test/test_turtle.py

```diff
@@ -54,6 +56,24 @@
 """
 
 
+@contextmanager
+def patch_screen():
+    """Patch turtle._Screen for testing without a display.
+
+    We must patch the _Screen class itself instead of the _Screen
+    instance because instantiating it requires a display.
+    """
+    m = unittest.mock.MagicMock()
+    m.__class__ = turtle._Screen
+    m.mode.return_value = "standard"
+
+    patch = unittest.mock.patch('turtle._Screen.__new__', return_value=m)
+    try:
+        yield patch.__enter__()
+    finally:
+        patch.__exit__(*sys.exc_info())
```

Comment:
It's a bit unusual to call the context manager protocol manually, also part of the context manager protocol handling is missing here.

```suggestion
    with unittest.mock.patch(
        'turtle._Screen.__new__',
        **{
            "return_value.__class__": turtle._Screen,
            "return_value.mode.return_value": "standard",
        },
    ) as mock_screen_new:
        yield mock_screen_new
```

Reply:
Thanks for spotting this! We've updated the `patch_screen` function with your suggestion 🙂